src/test/java/apoc/algo/WeaklyConnectedComponentsTest.java

```diff
@@ -0,0 +1,126 @@
+package apoc.algo;
+
+import apoc.util.TestUtil;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Map;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.test.TestGraphDatabaseFactory;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class WeaklyConnectedComponentsTest {
+	
+	private static GraphDatabaseService db;
+	// TODO refactor in multiple execute statements dirty workaround
+	public static final String CC_GRAPH =
+            "CREATE (:Node:Record { name : 'A' })  "+
+            "WITH count(*) as dummy " +
```

Comment:
you don't need the WITH's
FOREACH (name IN ['A',....'Z'] | CREATE (:Node:Record {name:name}))